test/e2e/serial/tests/workload_placement.go

```diff
@@ -1049,19 +1049,7 @@ func allocatableResourceType(nrtInfo nrtv1alpha1.NodeResourceTopology, resName c
 // NUMA nodes so that all the replicas can successfully obtain resources
 // from all the NUMA nodes.
 func leastAvailableResourceQtyInAllZone(nrtInfo nrtv1alpha1.NodeResourceTopology, baseload baseload.Load, resName corev1.ResourceName) resource.Quantity {
-	var maxResAllocatable resource.Quantity
-
-	// Finding the maximum allocatable resources of a resource type across all zones
-	for _, zone := range nrtInfo.Zones {
-		zoneQty, ok := e2enrt.FindResourceAllocatableByName(zone.Resources, resName.String())
-		if !ok {
-			continue
-		}
-		if zoneQty.Cmp(maxResAllocatable) > 0 {
-			maxResAllocatable = zoneQty
-		}
-	}
-
+	maxResAllocatable := e2enrt.GetMaxAllocatableResourceNumaLevel(nrtInfo, resName)
```

Comment:
the intent is good, but when a helper is used only once, we should generally not create it and inline it. The only noteworthy exception is the code extraction makes the code significantly more testable or significantly easier to read.
In this case we have some readability gains and in the followup patch you use it, which passes the code extraction threshold.

Next time please mention in the commit message you plan to use the extracted code and mention the reasons that made you decide to perform the extraction - the fact you extracted the code is evident from the change itself, the commit message should explain why and how, not what.

Reply:
I agree, thanks for highlighting this! I'll consider it next time